docs: Cleanup README grammar, punctuation, and capitalization (#654)
dbt-athena/dbt-athena: README.md

- `spark_cross_account_catalog` (`default=false`)
  - In Spark, you can query the external account catalog and for that the consumer account has to be configured to

>    access the producer catalog.

Author: I find this sentence extremely unclear. Can anyone help me try to parse/clarify what it means so I can rewrite it?

Reviewer: I would say something like this:

> When using Spark Athena workgroup, queries can only be run against catalogs located on the same AWS account. But sometimes you want to make queries to an other catalog located on an external AWS account. This requires an additional parameter on the spark properties setup. Set it to true when you want to query external catalogs.  If this flag is set to true, you can use the syntax `external_catalog_id/database.table` to access the external table on the external catalog (ex: 999999999999/mydatabase.cloudfront_logs` where 999999999999 is the external catalog ID)

And I will remove the next line

Author: Fixed in e5ac7db!

- Python models cannot
 [reference Athena SQL views](https://docs.aws.amazon.com/athena/latest/ug/notebooks-spark.html).
- Third-party Python libraries can be used, but they must be [included in the pre-installed list][pre-installed list]

>  or [imported manually][imported manually].

Author: I misspecified these links in #643. My mistake!

## Contracts

The adapter partly supports contract definitions:

- `data_type` is supported but needs to be adjusted for complex types. Types must be specified
  entirely (for instance `array<int>`) even though they won't be checked. Indeed, as dbt recommends, we only compare
  the broader type (array, map, int, varchar). The complete definition is used in order to check that the data types
  defined in Athena are ok (pre-flight check).

> - The adapter does not support the constraints since no constraints don't exist in Athena.

Reviewer: ```suggestion
- The adapter does not support the constraints since there is no constraint concept in Athena.
```
